spec/src/main/asciidoc/chapters/data-binding.asciidoc

```diff
@@ -0,0 +1,165 @@
+
+Let's have a look at the following resource for an example:
+
+[source,java,numbered]
+----
+@Path("form")
+public class FormResource {
+
+    @FormParam("age") 
+    @Min(18) 
+    private int age;
+
+    @POST
+    public Response handlePost() {
+        // ...
+    }
+}
+----
+
+This resource uses a `@FormParam` annotation to bind the value of the `age` form parameter to a resource field.
+It also uses the Bean Validation annotation `@Min` to specify a constraint on the value.
+
+When JAX-RS binds the submitted data to the field, two types of errors are possible:
+
+[horizontal]
+Binding Error:: 
+  This type occurs if JAX-RS is unable to convert the submitted value into the desired target Java type. 
+  For the resource shown above, such an error will be thrown if the user submits some arbitrary string like `foobar` which cannot be converted into an integer.
+Validation Error::  
+  If the submitted value can be converted into the target type, JAX-RS will validate the data according to the Bean Validation constraints. 
+  In our example submitting the value 16 would be considered invalid and therefore result in a constraint violation.
+  
+Unfortunately the JAX-RS data binding mechanism doesn't work well for web applications:
+
+  * Both binding and validation errors will cause JAX-RS to throw an exception which can only be handled by an `ExceptionMapper`. 
+    Especially JAX-RS won't execute the resource method if errors were detected.
+    This is problematic, because typically web applications will want to display the submitted form again and include a message explaining why the submission failed.
+    Implementing such a requirement using an `ExceptionMapper` is not feasible.
+  * The JAX-RS data binding is not locale-aware. 
+    This is a problem especially for numeric data types containing fraction digits (like `double`, `float`, `BigDecimal`, etc). 
+    By default, JAX-RS will always assume the US number format.
+    
+[[mvc_binding_annotation]]
+=== @MvcBinding annotation
+
+MVC addresses the shortcomings of the standard JAX-RS data binding by providing a special data binding mode optimized for web applications. 
+[tck-testable tck-id-mvc-binding]#You can enable the MVC specific data binding by adding a `@MvcBinding` annotation to the corresponding controller field or method parameter#.
+
+The following example shows a controller which uses a MVC binding on a controller field.
+
+[source,java,numbered]
+----
+@Controller
+@Path("form")
+public class FormController {
+
+    @MvcBinding
+    @FormParam("age") 
+    @Min(18) 
+    private int age;
+
+    @POST
+    public String processForm() {
+        // ...
+    }
+}
+----
+
+Please note that usually `@MvcBinding` will be used with `@FormParam` bindings, as they are very common in web application.
+However, depending on the specific use case, it may also be useful to use it with `@QueryParam` and other binding annotations.
```

Comment:
Why the limitation to `@FormParam` at all? `/articles?maxPrice=500` is not a specific use case ;)

Reply:
Good point. Back then I was actually not sure if people will user MVC binding with query parameter, but after thinking about this more, I guess it is a perfect use case. I adjusted the text accordingly.